nexus/db-queries/src/db/datastore/oximeter.rs

```diff
@@ -48,10 +65,18 @@ impl DataStore {
     ) -> Result<(), Error> {
         use db::schema::oximeter::dsl;
 
-        // If we get a conflict on the Oximeter ID, this means that collector instance was
-        // previously registered, and it's re-registering due to something like a service restart.
-        // In this case, we update the time modified and the service address, rather than
-        // propagating a constraint violation to the caller.
+        // If we get a conflict on the Oximeter ID, this means that collector
+        // instance was previously registered, and it's re-registering due to
+        // something like a service restart. In this case, we update the time
+        // modified and the service address, rather than propagating a
+        // constraint violation to the caller.
+        //
+        // TODO(john) - Should we ignore time_deleted or explicitly set it to
+        // NULL? We _shouldn't_ be called by an Oximeter that's been deleted,
+        // and if we are that indicates a bug somewhere else in the system
+        // (probably in reconfigurator or its related tasks). For now, we refuse
+        // to resurrect a deleted Oximeter: if we're called with a deleted
+        // instance, we'll leave it deleted.
```

Comment:
I don't think we should set `time_deleted` to NULL here.  If it were easy to do, I'd have this fail and return that back out to the Oximeter.  I don't think it's worth investing much effort in identifying this case that shouldn't happen though.